Classes/Utility/PathUtility.php

```diff
@@ -38,19 +38,26 @@ class PathUtility {
 	 * @return mixed
 	 */
 	public static function translatePath($path) {
-		if (is_array($path) == FALSE) {
-			$path = (0 === strpos($path, '/') ? $path : GeneralUtility::getFileAbsFileName($path));
+		if (is_string($path)) {
+			$path = GeneralUtility::isAbsPath($path) ? $path : GeneralUtility::getFileAbsFileName($path);
 			if (is_dir($path)) {
 				$path = realpath($path) . '/';
 			}
-			$path = GeneralUtility::fixWindowsFilePath($path);
-		} else {
-			foreach ($path as $key => $subPath) {
-				if (TRUE === in_array($key, self::$knownPathNames)) {
-					$path[$key] = self::translatePath($subPath);
-				}
-			}
+			return GeneralUtility::fixWindowsFilePath($path);
+		} elseif (is_array($path)) {
+			return array_map(array(__CLASS__, __METHOD__), $path);
 		}
-		return $path;
+		return NULL;
+	}
+
+	/**
+	 * Get fields from view configuration that are template paths
+	 *
+	 * @param array $viewConfiguration
+	 * @return array
+	 */
+	public static function getTemplatePathFields(array $viewConfiguration) {
```

Comment:
Sorry for the inconvenience. The original problem was that the method was not working 100% on `templateRootPaths/partialRootPaths/...`. Before we go full circle give me one last chance to keep the original behavior and only fix the little bug with that additional method.